services/whatpulse/whatpulse.tester.js

```diff
@@ -0,0 +1,30 @@
+import Joi from 'joi'
+import { createServiceTester } from '../tester.js'
+import { isOrdinalNumber } from '../test-validators.js'
+export const t = await createServiceTester()
+
+t.create('WhatPulse user as user id, category not from Ranks')
+  .get('/user/179734.json?category=DOWNLOAD')
+  .expectBadge({ label: 'WhatPulse DOWNLOAD', message: Joi.string() })
+
+t.create('WhatPulse user as user name, category from Ranks')
+  .get('/user/jerone.json?category=rankS/UplOad')
+  .expectBadge({ label: 'WhatPulse rankS/UplOad', message: isOrdinalNumber })
```

Comment:
The left side of the badge should be lowercase. Also, I don't think we need to mention WhatPulse here. Occasionally we need to include the name of the service to explain what the thing is, but if a metric is platform-agnostic we exclude the service name. So for example code coverage is an established metric. Whether we are measuring it on codecov or coveralls, 80% coverage is 80% coverage. We know what it means for a build to pass or fail. Whether it passed or failed on travis or circle isn't the key point. I think all these metrics (minutes of uptime, bytes downloaded, etc) are generic - 1,000 clicks is 1,000 clicks whatever service we use to measure it. There is some guidance on this in https://github.com/badges/shields/blob/7d452472defa0e0bd71d6443393e522e8457f856/CONTRIBUTING.md#badge-guidelines

Reply:
Ok, changed to: 
```js
label: `${metric}`,
```

What should be the defaultBadgeData , that will be the label in case of an error?
Currently I have:
```js
  static defaultBadgeData = { label: 'WhatPulse' }
```